pkg/lib/cockpit.js

```diff
@@ -1483,17 +1483,27 @@ function factory() {
             });
     };
 
-    function format_units(number, suffixes, factor, options) {
-        // backwards compat: "options" argument position used to be a boolean flag "separate"
-        if (!is_object(options))
-            options = { separate: options };
+    function format_units(suffixes, number, second_arg, third_arg) {
+        let options = second_arg;
 
+        /* compat API: we used to accept 'factor' as a separate second arg */
+        if (third_arg || (second_arg && !is_object(second_arg))) {
```

Comment:
We could add a console.warn() here about the deprecated API?

Reply:
Sure.

Reply:
Good point -- you can filter in the browser, and the console coalesces *adjacent* repeated messages, but not if there's other messages in between. So I suppose ideally we should have a `warned` global to remember, or forget about it and convert the world to .ts :grin: